src/galaxy/StarSystemGenerator.cpp

```diff
@@ -705,6 +705,9 @@ void StarSystemRandomGenerator::PickPlanetType(SystemBody *sbody, Random &rand)
 		// Anything bigger than 200 EU masses is a Gas Giant or bigger but the density changes to decrease from here on up...
 		sbody->m_radius = fixed::FromDouble(22.6 * (1.0 / pow(sbody->GetMassAsFixed().ToDouble(), double(0.0886))));
 	}
+	// randomize radius or the surface gravity will be consistently 1.0 g for planets in the 1 - 200 EU range.
+	sbody->m_radius = fixed::FromDouble(sbody->GetRadiusAsFixed().ToDouble() * ( 1.2 - (0.4 * rand.Double()))); // +/-20% radius
+
```

Comment:
Quick test. It didn't improve anything in the game. I think it could be an interesting idea for a future rewrite but here and now it's just going to make the code messier with no apparent gain.

I stuck this line in there after metallicity was set.

```cpp
	// randomize radius to take metallicity into account
	sbody->m_radius = fixed::FromDouble(sbody->GetRadiusAsFixed().ToDouble() *
 		( 1.1 - (0.2 * sbody->GetMetallicity()))); // +/-10% radius
```

Reply:
If we're going to do that, it'd probably be better to tie it all the way into body formation/distribution; e.g. minor moons / outer planets might be more icy than rocky which would affect their mass and radius together. Either mass + random density could inform metallicity, or pre-calculated metallicity could inform mass + density.

Reply:
@Web-eWorks good idea but probably out of scope for this PR

Reply:
I had to redo the tests above as I had borked the volume/density calculations. The conclusion is the same. It'll take some more consideration to rewrite the planet generation.

> or pre-calculated metallicity could inform mass + density.

This is what I'm curious to look into first. I'm thinking of summing up my thoughts on this in a forum post.